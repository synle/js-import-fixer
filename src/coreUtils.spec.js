const coreUtils = require("./coreUtils");

describe("coreUtils.js", () => {
  describe("getAliasName", () => {
    test("complex moduleName with as", async () => {
      const actual = coreUtils.getAliasName("MySimpleLibrary as MyAliasName");
      expect(actual).toBe("MyAliasName");
    });

    test("simple moduleName", async () => {
      const actual = coreUtils.getAliasName("MySimpleLibrary");
      expect(actual).toBe("MySimpleLibrary");
    });
  });

  describe("getModuleName", () => {
    test("complex moduleName with as", async () => {
      const actual = coreUtils.getModuleName("MySimpleLibrary as MyAliasName");
      expect(actual).toBe("MySimpleLibrary");
    });

    test("simple moduleName", async () => {
      const actual = coreUtils.getModuleName("MySimpleLibrary");
      expect(actual).toBe("MySimpleLibrary");
    });
  });

  // test("getLibrarySortOrder", async () => {
  //   // TODO
  // });
});
