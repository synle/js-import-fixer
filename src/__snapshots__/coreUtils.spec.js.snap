// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`coreUtils.js process sample_1.js simple 1`] = `
"import {aliasMethodLib1 as myAliasMethod1} from 'externalLib1';
import {constant1} from 'externalLib1';
import {methodLib1} from 'externalLib1';
import externalLib1 from 'externalLib1';
import {constant2} from 'externalLib2';
import externalLib2 from 'externalLib2';
var a1 = constant1;
methodLib1();
externalLib1();
myAliasMethod1();

var a2 = constant2;
var temp2 = externalLib2();"
`;

exports[`coreUtils.js process sample_1.js withGroupImport 1`] = `
"import externalLib1, { aliasMethodLib1 as myAliasMethod1, constant1, methodLib1 } from 'externalLib1';
import externalLib2, { constant2 } from 'externalLib2';
var a1 = constant1;
methodLib1();
externalLib1();
myAliasMethod1();

var a2 = constant2;
var temp2 = externalLib2();"
`;

exports[`coreUtils.js process sample_2.js simple 1`] = `
"import externalLib1, { aliasMethodLib1 as myAliasMethod1, constant1, methodLib1 } from 'externalLib1';
import externalLib2, { constant2 } from 'externalLib2';
var a1 = constant1;
methodLib1();
externalLib1();
myAliasMethod1();

var a2 = constant2;
var temp2 = externalLib2();"
`;

exports[`coreUtils.js process sample_2.js withGroupImport 1`] = `
"import externalLib1, { aliasMethodLib1 as myAliasMethod1, constant1, methodLib1 } from 'externalLib1';
import externalLib2, { constant2 } from 'externalLib2';
var a1 = constant1;
methodLib1();
externalLib1();
myAliasMethod1();

var a2 = constant2;
var temp2 = externalLib2();"
`;
